app/api/v1/farmers.py

```
@@ -0,0 +1,82 @@
+async def create_farmer_profile(
+    user_id: int,
+    farmer_data: FarmerCreate,
+    db: AsyncSession = Depends(get_db)
+):
+    """Create a new farmer profile."""
+    farmer_service = FarmerService(db)
+    try:
+        farmer = await farmer_service.create_farmer_profile(user_id, farmer_data)
+        return farmer
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)